### PR TITLE
Enable Go workspace

### PR DESCRIPTION
**Pull request by csweichel**
## Description
This PR enables Go's new workspace feature for this repo. To make this happen we had to add Go workspace support to leeway first.

[Go Workspaces at Gitpod - Watch Video](https://www.loom.com/share/678cc0ffc9d640bdaaaea87094ef9ab2)

<a href="https://www.loom.com/share/678cc0ffc9d640bdaaaea87094ef9ab2">
    <p>Go Workspaces  at Gitpod - Watch Video</p>
    
  </a>

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by meysholdt — APPROVED** [review]
🚀

**Comment by easyCZ** [comment]
Open this PR with Goland, and this is the first time that I did not have to manually pull dependencies to get Goland to see them. Yes, this time it took a considerable time to index (as we don't prebuild indices on PRs) but one main it should be fast, like really fast.

Looks great, thanks @csweichel!

**Review by jeanp413 — APPROVED** [review]

**Review by easyCZ — APPROVED** [review]

File: components/docker-up/docker-up/runc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+This file is downloaded as part of the build.
```
Comment (sagor999) [inline components/docker-up/docker-up/runc:1]: is this due to new go workspace leeway setup?

**Review by sagor999 — APPROVED** [review]

File: components/docker-up/docker-up/runc (hunk 1 above)
Comment (csweichel) [reply components/docker-up/docker-up/runc:1]: It's mainly for convenience. Without those files present, we'd see an (inconsequential) error when opening a workspace.

File: components/docker-up/docker-up/main.go [hunk 2]
```diff
@@ -377,8 +377,6 @@ func installDocker() error {
 		//nolint:errcheck
 		os.Chtimes(dstpath, hdr.AccessTime, hdr.ModTime)
 	}
-
-	return nil
```
[the remark this answers is not in the document]
Comment (csweichel) [reply components/docker-up/docker-up/main.go:381]: Wasn't actually reachable. The `for` loop ends with a return in all cases.
The linter complained about this.

**Review by akosyakov — APPROVED** [review]
lgtm

**Review by nandajavarma — APPROVED** [review]

**Review by geropl — APPROVED** [review]
:tada:

**Pull request merged**